src/app/features/tech-record/components/tech-record-amend-vrm/tech-record-amend-vrm.component.ts

```diff
@@ -63,12 +65,10 @@ export class AmendVrmComponent implements OnInit {
     return FormNodeWidth.L;
   }
 
-  ngOnInit(): void {
-    if (!this.techRecord) {
-      this.navigateBack();
-    }
-
-    this.actions$.pipe(ofType(updateTechRecordsSuccess), take(1)).subscribe(() => this.navigateBack());
+  get vehicleType(): VehicleTypes | undefined {
+    return this.techRecord?.vehicleType === VehicleTypes.TRL && this.techRecord.euVehicleCategory === EuVehicleCategories.O1
+      ? VehicleTypes.SMALL_TRL
+      : this.techRecord?.vehicleType;
```

Comment:
I would suggest putting this function to get the vehicle type in the tech record service. Ideally we would be removing this "dodgy" behaviour at some point, and this separation of concerns/abstraction would probably help that process. Plus it means less repeated code seeing as I've seen this in a few components already.

Reply:
Suggestion applied.